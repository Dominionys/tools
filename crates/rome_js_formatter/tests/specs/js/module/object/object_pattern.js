// Break over multiple lines if it contains a sub pattern
const {b: {test}} = test;

// non-shorthand and whole statement exceeds line width
const { id, isStatic: isStatic, method, methodId, g } = privateNamesMap.get(name);

// default and whole statement exceeds line width
const { id, isStatic = true, method, methodId, gee } = privateNamesMap.get(name);

// sub pattern
const { id, isStatic: {sub}, method, methodId, gee } = privateNamesMap;


const { id, //middle comment
  isStatic, method, methodId, gee } = privateNamesMap;

try {
  // you should not break
} catch ({ data: { message }}) {

}