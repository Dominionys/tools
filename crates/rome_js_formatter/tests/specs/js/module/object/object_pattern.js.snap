---
source: crates/rome_js_formatter/tests/spec_test.rs
assertion_line: 223
expression: object_pattern.js

---
# Input
// Break over multiple lines if it contains a sub pattern
const {b: {test}} = test;

// non-shorthand and whole statement exceeds line width
const { id, isStatic: isStatic, method, methodId, g } = privateNamesMap.get(name);

// default and whole statement exceeds line width
const { id, isStatic = true, method, methodId, gee } = privateNamesMap.get(name);

// sub pattern
const { id, isStatic: {sub}, method, methodId, gee } = privateNamesMap;


const { id, //middle comment
  isStatic, method, methodId, gee } = privateNamesMap;

try {
  // you should not break
} catch ({ data: { message }}) {

}
=============================
# Outputs
## Output 1
-----
Indent style: Tab
Line width: 80
Quote style: Double Quotes
-----
// Break over multiple lines if it contains a sub pattern
const {
	b: { test },
} = test;

// non-shorthand and whole statement exceeds line width
const {
	id,
	isStatic: isStatic,
	method,
	methodId,
	g,
} = privateNamesMap.get(name);

// default and whole statement exceeds line width
const {
	id,
	isStatic = true,
	method,
	methodId,
	gee,
} = privateNamesMap.get(name);

// sub pattern
const {
	id,
	isStatic: { sub },
	method,
	methodId,
	gee,
} = privateNamesMap;

const {
	id, //middle comment
	isStatic,
	method,
	methodId,
	gee,
} = privateNamesMap;

try {
	// you should not break
} catch ({ data: { message } }) {}

